ftplugin/markdown/quicklink.vim

```diff
@@ -95,4 +95,45 @@ function! OpenLinkOnCurrentLine()
   call system('open ' . expand('<cWORD>'))
 endfunction
 
+function! s:OnMarkdownLink()
+    return synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkText" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "markdownLinkTextDelimiter" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdLink" ||
+                \ synIDattr(synID(line("."),col("."),1),"name") == "mkdDelimiter"
+endf
+
+function! s:OpenMarkdownLink()
+  "get the initial position of the curosr
+  let initial_pos = getpos('.')
+  "empty the @a register in the case cursor is not well palced and that old
+  "patterns would still trigger.
+  let @a = ""
+  normal "ayi]
+  let escaped_link_name = escape(getreg('a'), '&')
+  let link_target_pattern = '\v^\['.escaped_link_name.'\]: (%(ftp[s]?|http[s]?):\/\/\S+)>'
+  " go to link location
+  let found = search(link_target_pattern, 'e')
```

Comment:
I think this is clearer is you inline the search call into the `if`, so it would just be:

``` vim
if search(link_target_pattern, 'e')
```